src/core/thread/mle_router.cpp

```diff
@@ -3334,7 +3334,7 @@ ThreadError MleRouter::RestoreChildren(void)
     for (uint8_t i = 0; ; i++)
     {
         Child *child;
-        otChildInfo childInfo;
+        ChildInfo childInfo;
         uint16_t length;
 
         length = sizeof(childInfo);
```

Comment:
I have updated otPlatSettingsGet() to be consistent with its documentation: return the actual length of the setting, but not the actual length it read to aValue. 
And I also updated the length check logic to support backward compatibility. Please have review again, thanks.